pow-migration/src/main.rs

```diff
@@ -164,9 +173,56 @@ async fn main() {
                 std::process::exit(1);
             }
         };
-        println!("Registered validators:");
-        for validator in registered_validators {
-            println!("{}", validator.validator.validator_address);
+
+        if args.list_validators {
+            println!("Registered validators:");
+            for validator in registered_validators {
+                println!("{}", validator.validator.validator_address);
+            }
+        } else {
+            if pow_client.block_number().await.unwrap()
+                < block_windows.pre_stake_end + block_windows.block_confirmations
+            {
+                log::error!("The pre-staking window is not closed yet, generating the list is not possible at this time.");
+                std::process::exit(1);
+            }
+
+            let pre_stakers = match get_stakers(
+                &pow_client,
+                &registered_validators,
+                block_windows.pre_stake_start..block_windows.pre_stake_end,
+            )
+            .await
+            {
+                Ok((pre_stakers, _)) => match args.validator {
+                    Some(address) => {
+                        if let Ok(address) = Address::from_any_str(&address) {
+                            pre_stakers
+                                .into_iter()
+                                .filter(|pre_staker| pre_staker.delegation == address)
+                                .collect()
+                        } else {
+                            log::error!("Invalid address as --validator argument provided");
+                            std::process::exit(1);
+                        }
+                    }
+                    None => pre_stakers,
+                },
+                Err(error) => {
+                    log::error!(?error, "Couldn't get pre-stakers list");
+                    std::process::exit(1);
+                }
+            };
+
+            println!("Pre-stakers:");
+            for pre_staker in pre_stakers {
+                println!(
+                    "{} has delegated stake to {} with a prestake of {} NIM",
+                    pre_staker.staker_address,
+                    pre_staker.delegation.to_user_friendly_address(),
```

Comment:
Display of an address should already use the user friendly representation
```suggestion
                    pre_staker.delegation,
```